src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -0,0 +1,328 @@
+       key  = os.path.join('certs',config['sas_hareness'][N]['key_file'])
+       fad_object_th.append(FullActivityDump(cert,key,config['sas_hareness'][N]['url']))
+       fad_records.append(fad_object_th[N])
+
+    #for UUT creating fad object
+    cert = os.path.join('certs',config['sas_uut']['cert_file'])
+    key  = os.path.join('certs',config['sas_uut']['key_file'])
+    fad_uut_object = FullActivityDump(cert,key,config['sas_uut']['url'])
+    fad_records.append(fad_uut_object)
+
+    # informing SAS UUT about SAS TH 
+    for N in len(config['sas_hareness']):
+       self._sas_admin.InjectPeerSas({'certificateHash': config[N]['certificateHash'],
+                                     'url': 'https://' + config[N] ['url']+ '/dump'}) 
+    
+
+    #Step 4,5 : Inject IAP protected entities into UUT
+    for k in  config['K']:	
+       for records in config['IAPRecords'][k]:
+          if records['type'] == "FSS":
+             self._sas_admin.InjectFss(records['Record'])
+          if records['type'] == "GWPZ":
+             self._sas_admin.InjectWisp(records['Record'])
+          if records['type'] == "ESC":
+             self._sas_admin.InjectEsc(records['Record'])
+          if records['type'] == "PPA":
+             self._sas_admin.InjectZoneData(records['Record'])
+
+     
+        
+       # Step 6 : Trigger FAD generation with SAS test harness
+       if len(config['sas_hareness']) > 0 :
+          self._sas_admin.TriggerFullActivityDump() # triggers to UUT
+
+          # Step 7 : Pull FAD records from SAS UUT
+          fad_uut_object.initiate_full_activity_dump()
+          
+          for N in len(config['sas_hareness']):
+             fad_object_th[N].initiate_full_activity_dump()
+
+
+
+       # Step 8 : Trigger CPAS and wait until completion
+       self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+       # Step 9 : Call IAP reference model, calculated at step 12
+       #collect all the protected Entites
+       protectedEntities.append(config['IAPRecords'][k])
+
+       for pe in protectedEntities:
+          fad_records_iap = []
+          if pe['type'] == 'FSS':
+           
+             # Find the FSS records with OOBE protection
+             fss_with_oobe = self.findFSSInIAPwithOOBE(protectedEntities)
+             if fss_with_oobe :
+                # Call FSS purge list reference model to fetch cbsd to be purged. output:List of CBSDs
+                cbsd_purge_list = FSSPurgeListRefModel(fad_records,fssOobe)
+ 
+                # Remove purged cbsd from fad record
+                fad_records_iap = removePurgeCbsd(fad_records,cbsd_purge_list)
+             else:
+                fad_records_iap = fad_records
+          #Call IAP reference model, fad_records contains fad_UUT and fad_SAS_TH
+          #To-Do : If there is any change in the API defination, parameters need to be updated
+          iap_refmodel_margin_list = IapReferenceModel(config['headroom'], pe,fad_records_iap)
```

Comment:
*All* of the pre-IAP filtering needs to be done before you can run the IAP reference model for any protected entity.

Reply:
Added,   Invoked the pre-IAP API